arch/risc-v/src/common/riscv_macros.S

```diff
@@ -227,8 +222,8 @@
   REGLOAD      t0, REG_INT_CTX(\out)
   li           t1, MSTATUS_FS
   and          t2, t0, t1
-  li           t1, MSTATUS_FS_INIT
-  ble          t2, t1, 1f
```

Comment:
This logic is correct, you need to restore when status in {DIRTY, CLEAN}

Reply:
Yes, I think so.